FlowCrypt/Controllers/Msg/MessageViewController.swift

```diff
@@ -266,7 +265,16 @@ extension MessageViewController {
     }
 
     @objc private func handleMailTap() {
-        showToast("Marking as unread will be implemented soon")
+        messageOperationsProvider.markAsUnread(message: input.objMessage, folder: input.path)
+            .then(on: .main) { [weak self] in
+                guard let self = self else { return }
+                self.input.objMessage = self.input.objMessage.markAsRead(false)
+                self.onCompletion?(MessageAction.markAsUnread, self.input.objMessage)
```

Comment:
I suppose `MessageAction.markAsUnread` could be updated to `MessageAction.changeReadFlag` ? Since it's now used for both marking as read and unread? Unless I'm wrong?